### PR TITLE
configure: Don't succeed if valgrind isn't available. Fixes #18588

### LINKED ISSUE #18588: run-pass-valgrind tests fail when valgrind doesn't exist
I get `task '<main>' panicked at 'Can't find Valgrind to run Valgrind tests', /opt/dev/rust3/src/compiletest/compiletest.rs:44` when running the test suite now. These tests should probably be disabled when configure doesn't find valgrind.

### PR DESCRIPTION
**Pull request by brson**
Based on @emanueLczirai's patch.

r? @nrc

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -750,6 +750,15 @@ then
     fi
 fi
 
+# By default the test suite *requires* valgrind. Detect it's absence.
```
Comment (apasel422) [inline configure]: s/it's/its/

**Comment by nrc** [comment]
r+, I guess with the typo fixed

**Comment by Manishearth** [comment]
Can valgrind be disabled by default in the configure? IMO `./configure` should generally just work on common systems, and valgrind is usually not there out of the box.

**Comment by Manishearth** [comment]
```
configure: error: valgrind not found, but needed. consider adding --disable-valgrind-rpass
program finished with exit code 1
elapsedTime=2.558000
```

**Comment by emanueLczirai** [comment]
One thing to note is that `--disable-valgrind` doesn't imply `--disable-valgrind-rpass`  as far as I remember. It seemed to me, it was intended this way from the beginning.
The relevant "discussion" is here: https://github.com/rust-lang/rust/issues/18588#issuecomment-71195350

**Comment by steveklabnik** [comment]
@brson: now that https://github.com/rust-lang/rust/pull/24859 is merged, is this PR needed?

**Comment by brson** [comment]
I think it is not.

**Pull request closed without merging**